great_expectations/core/expectation_suite.py

```diff
@@ -161,6 +161,11 @@ def __eq__(self, other):
         if not isinstance(other, self.__class__):
             # Delegate comparison to the other instance's __eq__.
             return NotImplemented
+
+        # sort expectations to be deterministic
+        self.expectations.sort()
```

Comment:
@Shinnnyshinshin I feel that we should use `sorted()` because it does not modify the original list.  Do we really want to modify the `self.expectations` list?..

Reply:
Great point. After discussion, we have determined that this sorting step is not necessary. PR has been updated to reflect this clearer understanding.